process/smartContract/process.go

```diff
@@ -1550,6 +1573,10 @@ func (sc *scProcessor) penalizeUserIfNeeded(
 		return
 	}
 
+	if !sc.shouldCheckValues() {
```

Comment:
I do not really like this check. This is not needed. TxSimulate is actual simulation of ones transaction - with fees and everything.

Reply:
VMQuery is a thing where you might do these without check.

Reply:
removed.